tests/Temporalio.Tests/Worker/WorkflowWorkerTests.cs

```diff
@@ -4789,6 +4790,215 @@ public async Task ExecuteWorkflowAsync_CoroutinesAfterComplete_GetProcessed()
         await flagReplayer.ReplayWorkflowAsync(historyWithFlag);
     }
 
+    public class CallbackActivities
+    {
+        private readonly Func<string, Task<string>> func;
+
+        public CallbackActivities(Func<string, Task<string>> func) => this.func = func;
+
+        [Activity]
+        public Task<string> DoActivityAsync(string param) => func(param);
+    }
+
+    [Workflow]
+    public class SemaphoreWorkflow : IDisposable
+    {
+        private readonly SemaphoreSlim semaSlim = new(1, 1);
+        private readonly Semaphore semaHeavy = new(1, 1);
+
+        [WorkflowRun]
+        public Task RunAsync() => Workflow.WaitConditionAsync(() => CompletedUpdates.Count == 3);
+
+        [WorkflowQuery]
+        public List<string> CompletedUpdates { get; } = new();
+
+        [WorkflowUpdate]
+        public async Task UpdateWithAsyncSlim()
+        {
+            await semaSlim.WaitAsync();
+            try
+            {
+                CompletedUpdates.Add(await Workflow.ExecuteActivityAsync(
+                    (CallbackActivities act) => act.DoActivityAsync(Workflow.CurrentUpdateInfo!.Id),
+                    new() { StartToCloseTimeout = TimeSpan.FromMinutes(10) }));
+            }
+            finally
+            {
+                semaSlim.Release();
+            }
+        }
+
+        [WorkflowUpdate]
+        public async Task UpdateWithSyncSlim()
```

Comment:
Curious why somebody would use this approach in an async handler.

Reply:
They wouldn't, this will deadlock. The test is just confirming these sync calls are no good.